text/0000-add-limits-trait.md

```diff
@@ -0,0 +1,133 @@
+- Feature Name: Limits trait for the rust types
+- Start Date: 2017-12-20
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+This is an RFC to add a universal trait for the type limits.
+
+# Motivation
```

Comment:
The language in this section could be improved a bit =)

Reply:
Unfortunately I am not an english native speaker. Could you correct me please?

Reply:
I'd write it as (with some slight semantic changes and clarifications and added Haskell as an example):

> The motivation is simple: By providing the methods in a trait, user code is able to require with a bound `X: Limit<Y>` that `X` has certain limits of type `Y`, which enables generic reasoning and simplifies code.
>
> Another motivation is that we already have inherent methods [`.max_value()`](https://doc.rust-lang.org/1.22.1/std/?search=max_value) and [`.min_value()`](https://doc.rust-lang.org/1.22.1/std/?search=min_value) on all primitive numeric types. Generalizing those methods as a trait simplifies the code and avoids duplication.
>
> Looking at other languages, C++ provides [`std::numeric_limits`](http://en.cppreference.com/w/cpp/types/numeric_limits), while Haskell provides the [`Bounded`](http://hackage.haskell.org/package/base-4.10.1.0/docs/Prelude.html#v:minBound) typeclass. This provides precedent that such a facility belongs in the standard library.